.changeset/eleven-forks-sleep.md

```diff
@@ -0,0 +1,18 @@
+---
+'xstate': major
+---
+
+The `self` actor reference is now available in all action and guard meta. This makes it easier to reference the "self" `ActorRef` so that actions such as `sendTo` can include it in the event payload:
```

Comment:
It's also passed to `delay` creators - this can be done to keep things the same/similar although I really can't find a reason to use `self` in there (and perhaps we could extend the argument against adding `self` to guards here, `delay`s could be treated as pure/idempotent).

It's not passed to `input` factories though - and that one should likely be possible. The parent should be able to pass its own reference to the child through `input`